test/src/api/responderTournamentTest.php

```diff
@@ -0,0 +1,434 @@
+
+        // A tournament with an invalid number of wins should fail
+        $this->verify_api_createTournament_failure(
+            array(), 'Tournament create failed because the maximum number of wins was invalid',
+            'Single Elimination', 4, 0
+        );
+
+        // A tournament with an invalid tournament type should fail
+        $this->verify_api_createTournament_failure(
+            array(), 'Invalid tournament type',
+            'foobar', 4, 3
+        );
+
+        // A tournament with an invalid number of players should fail
+        $this->verify_api_createTournament_failure(
+            array(), 'Invalid number of players for this tournament type',
+            'Single Elimination', 5, 3
+        );
+
+        // Create a tournament to use for successful tournament tests
+        $tournamentId = $this->verify_api_createTournament(
+            array(),
+            'Single Elimination', 4, 1
+        );
+
+        $expData = array(
+            'tournamentId' => $tournamentId,
+            'type' => 'SingleElimination',
+            'nPlayers' => 4,
+            'maxWins' => 1,
+            'description' => '',
+            'creatorDataArray' => array(
+                'creatorId' => $_SESSION['user_id'],
+                'creatorName' => $_SESSION['user_name'],
+            ),
+
+            'tournamentRoundNumber' => 1,
+            'tournamentState' => 'JOIN_TOURNAMENT',
+            'currentPlayerIdx' => FALSE,
+            'gameDataArrayArray' => array(),
+            'remainCountArray' => array(),
+            'timestamp' => NULL,
+            'isCreator' => TRUE,
+            'isWatched' => TRUE,
+            'maxRound' => 2,
+        );
+        $retval = $this->verify_api_loadTournamentData($expData, $tournamentId);
+
+        // data from the perspective of a non-participant
+        $_SESSION = $this->mock_test_user_login('responder004');
+        $expData['isCreator'] = FALSE;
+        $expData['isWatched'] = FALSE;
+        $retval = $this->verify_api_loadTournamentData($expData, $tournamentId);
+        $_SESSION = $this->mock_test_user_login('responder003');
+        $expData['isCreator'] = TRUE;
+        $expData['isWatched'] = TRUE;
+
+        // attempt an invalid tournament dismiss
+        $this->verify_api_dismissTournament_failure(
+            array(), "Only participants can dismiss tournaments",
+            $tournamentId
+        );
+        $retval = $this->verify_api_loadTournamentData($expData, $tournamentId);
+
+        // join the tournament
+        $_SESSION = $this->mock_test_user_login('responder004');
+        $this->verify_api_updateTournament(
+            array(), $tournamentId, 'join', array('Avis')
+        );
+        $expData['remainCountArray'] = array(0 => 0);
+        $expData['playerDataArray'] = array(
+            0 => array(
+                'playerId' => $_SESSION['user_id'],
+                'playerName' => $_SESSION['user_name'],
+                'buttonId' => 256,
```

Comment:
Why is the `buttonId` in this return value?  Does the UI need it?

AFAIK non-tournament game data never includes `buttonId` in button information, so i don't think we should include it in the tournament either.

Reply:
Happy to remove this, you're right, it's not necessary.